tests/manage/cluster/nodes/test_disk_failures.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import node
+from ocs_ci.framework.testlib import tier4, ignore_leftovers
+from ocs_ci.framework import config
+from ocs_ci.utility import aws
+from tests.manage.cluster.nodes.base_node_ops import BaseNodes
+
+logger = logging.getLogger(__name__)
+
+
+@tier4
+@pytest.mark.skipif(
+    condition=config.ENV_DATA['platform'] != 'AWS',
+    reason="Tests are not running on AWS deployed cluster"
+)
+@ignore_leftovers
+class TestDetachAttachWorkerVolumeAWS(BaseNodes):
+    """
+    Test class for detach and attach worker volume
+
+    """
+    workers = list()
+
+    def set_workers(self, num_of_workers):
+        """
+        Set workers for the test
+
+        Args:
+            num_of_workers (int): Number of workers needed for the test
+
+        """
+        nodes = node.get_node_objs()
+        workers = [
+            n for n in nodes if 'worker' in n.get('metadata')
+            .get('annotations').get('machine.openshift.io/machine')
+        ]
+        self.workers = workers[:num_of_workers]
+
+    @pytest.mark.polarion_id("OCS-1085")
+    def test_detach_attach_worker_volume(self, aws_obj, resources):
+        """
+        Detach and attach worker volume
+
+        """
+        self.set_workers(1)
+        worker = self.workers[0]
+
+        instance = aws.get_instances_ids_and_names([worker])
+
+        instance_id = list(instance)[0]
+        volume = aws.get_data_volumes(instance_id)[0]
+
+        ec2_volume = aws_obj.ec2_resource.Volume(volume['id'])
+
+        aws_obj.detach_volume(volume)
```

Comment:
bit confusing,
here you detach volume and few lines later attach ec2_volume.
comments will help a lot to understand what's the flow

Reply:
done